state-chain/runtime/src/chainflip/solana_elections.rs

```diff
@@ -224,8 +268,13 @@ impl
 					<SolanaEgressWitnessing as ElectoralSystem>::ElectionIdentifierExtra,
 				>,
 			>,
+			Vec<
+				ElectionIdentifier<
+					<SolanaVaultSwapTracking as ElectoralSystem>::ElectionIdentifierExtra,
+				>,
+			>,
 		),
-		_context: &Self::OnFinalizeContext,
+		context: &Self::OnFinalizeContext,
```

Comment:
👀 I'm about to delete the `context` for the hooks in my refactor PR - we can instead just get the block number directly since we're in the runtime.

Reply:
My first approach was to directly get the block number but couldn't do so. The problem is that we dont have access to the Runtime here since it is a standalone module and is not part of the Pallet impl which means it doesnt have access to the runtime generic ```T```

Reply:
you don't have access because it's not in a generic context, you don't need T, you can just use `Runtime`

Reply:
We dont have access to the concrete ```Runtime``` type either since we are at pallet level and the Runtime is constructed at the Runtime level (one level higher). Thats the problem. The way to get access to the Runtime at the pallet level is the Runtime generic ```T``` which we cant use here either because of the reason i mentioned.

Reply:
clarified in dms. I thought kyle was talking about the ```context``` in the ```ElectoralSystem``` trait whereas he was talking about the ```context``` in ```Hooks``` trait.